src/nvim/lua/executor.c

```diff
@@ -1432,3 +1433,35 @@ void nlua_execute_on_key(int c)
 #endif
 }
 
+void nlua_do_ucmd(ucmd_T *cmd, exarg_T *eap)
+{
+  lua_State *const lstate = global_lstate;
+
+  nlua_pushref(lstate, cmd->uc_luaref);
+
+  lua_newtable(lstate);
+  lua_pushboolean(lstate, eap->forceit == 1);
+  lua_setfield(lstate, -2, "bang");
+
+  lua_pushinteger(lstate, eap->line1);
+  lua_setfield(lstate, -2, "lnum");
+
+  lua_pushinteger(lstate, eap->line2);
+  lua_setfield(lstate, -2, "end_lnum");
+
+  lua_pushstring(lstate, (const char *)eap->arg);
+  lua_setfield(lstate, -2, "args");
+
+  // The sizes of this buffer is chosen empirically to be large enough to hold
+  // every possible modifier (with room to spare). If the list of possible
+  // modifiers grows this may need to be updated.
+  char buf[200] = { 0 };
+  (void)uc_mods(buf);
```

Comment:
wouldn't we want this to be a table with mods in the lua API? or just keys on the arg table directly, perhaps. Perhaps we can make `add_cmd_modifier` take a flag to add `mod_str` as a key instead, so we can still share the `uc_mods` code with the vim script implementation?

Reply:
I actually think leaving it as a string is better because in many cases `<mods>` is passed on directly to another command that also takes mods, so this wouldn't require any intermediate processing. But if for some reason a Lua function *does* need the mods in a table it's trivially easy to use `vim.split()`.

Reply:
ok, let's keep it like it is for now :)